### ISSUE TITLE
fn quickstart doesn't work.

### ISSUE THREAD

**Issue by nicolasnoble** [serverless/serverless#4979]
# This is a Bug Report

## Description
Following https://serverless.com/framework/docs/providers/fn/guide/quick-start/ simply does not work.

## TL;DR:

The quickstart should mention the fact the `FN_REGISTRY` environment variable is mandatory for proper execution.

## Additional Data
Here's what happens when one follows the quickstart instructions:

```
$ serverless deploy -v
Serverless: Packaging service...
Serverless: Excluding development dependencies...

  Type Error ---------------------------------------------

  Cannot read property 'endsWith' of undefined

     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com

  Your Environment Information -----------------------------
     OS:                     linux
     Node Version:           8.10.0
     Serverless Version:     1.27.2
```

This error is caused by a bug in the `serverless-fn` package, and is solved by https://github.com/fnproject/serverless-fn/pull/1.

Now, when one applies this patch locally to `node_modules/serverless-fn`, trying a deployment yields the following:
```
Pushing hello:0.0.1 to docker registry....
The push refers to repository [docker.io/library/hello]
ecc4f29369bc: Preparing
834c3bd43478: Preparing
3749eee0969f: Preparing
0b3e54ee2e85: Preparing
ad77849d4540: Preparing
5bef08742407: Waiting
denied: requested access to the resource is denied

  Error --------------------------------------------------

  failed to push hello:0.0.1
```

Turns out that the current way the code is designed, the `FN_REGISTRY` variable is in fact mandatory, even though it tries to make it optional. So after installing a local docker registry following the instructions in https://docs.docker.com/registry/deploying/, and setting `FN_REGISTRY` to `localhost:5000`, a deployment and invocation eventually works fine.

**Comment by HyperBrain** [comment]
@hibooboo2 @DavidWells Any idea, how to improve this?

**Comment by andrewsosa** [comment]
This issue is still not fixed, still getting the same `Cannot read property 'endsWith' of undefined`.

**Comment by nicolasnoble** [comment]
To be fair, I kind of gave up on this, due to the lack of reactivity. I realized a few things:

1) My pull request to fix these was issue number 1 in the serverless-fn repository in the fnproject organization.
2) The version number of it is 0.0.1
3) There has been exactly zero activity since the initial code drop a few months ago.

This doesn't give any good impression on how stable and maintained this project is. I'll leave the issue and pull requests opened, but I've moved on to openfaas due to this.

**Comment by RogerioDosSantos** [comment]
+1 same issue. Evaluating the best framework and the @nicolasnoble comments let me worry. Fn sounded very good particularly running on the local computer as it would be in the server. @nicolasnoble how is your experience with openfaas so far?

**Comment by nicolasnoble** [comment]
I was able to use openfaas in a satisfactory manner. It's well maintained and has good velocity. Deployment was a breathe. I eventually moved to kubernetes deployment, which makes the whole thing even better.

**Comment by Hokutosei** [comment]
I just hit this one today... any progress on here?

```
➜  apps serverless create --template fn-nodejs --path new-project
Serverless: Generating boilerplate...
Serverless: Generating boilerplate in "/Users/jeanepaul/Developer/fn/apps/new-project"
 _______                             __
|   _   .-----.----.--.--.-----.----|  .-----.-----.-----.
|   |___|  -__|   _|  |  |  -__|   _|  |  -__|__ --|__ --|
|____   |_____|__|  \___/|_____|__| |__|_____|_____|_____|
|   |   |             The Serverless Application Framework
|       |                           serverless.com, v1.40.0
 -------'

Serverless: Successfully generated boilerplate for template: "fn-nodejs"
➜  apps ls
new-project
➜  apps cd new-project
➜  new-project ls
hello          package.json   serverless.yml
➜  new-project npm install
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN new-project@1.0.0 No description
npm WARN new-project@1.0.0 No repository field.

added 106 packages from 119 contributors and audited 152 packages in 5.195s
found 1 low severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
➜  new-project ls
hello             node_modules      package-lock.json package.json      serverless.yml
➜  new-project serverless deploy -v
Serverless: Packaging service...
Serverless: Excluding development dependencies...

  Type Error ---------------------------------------------

  Cannot read property 'endsWith' of undefined

     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com

  Your Environment Information ---------------------------
     OS:                     darwin
     Node Version:           10.15.3
     Serverless Version:     1.40.0
```

**Comment by nicolasnoble** [comment]
Don't bother. There had been absolutely zero activity here since the initial code drop. Use openfaas or kubernetes.

**Comment by agustim** [comment]
If you check fnDeploy.js in serverless-fn module line 242 you can read:
```
     imageName(func) {
        let fname = func.name;
        if (!fname.includes('/')) {
		// then we'll prefix FN_REGISTRY
            let reg = process.env.FN_REGISTRY;
            if (reg !== '') {
                if (!reg.endsWith('/')) {
...
```

The problem: FN_REGISTRY is not in environment, you can test:

FN_REGISTRY=your-docker-hub-user sls deploy -v

Deploy build a docker image, and push to your docker-hub space, but no work, yet!
```
  Error --------------------------------------------------                                     
                                  
  Not Found                        
  
     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.       
  
  Stack Trace --------------------------------------------
```
I try run docker image with:
```
docker run you-docker-hub-user/hello:0.0.1
The Node.js FDK does not support the '' format, change the function format to 'http-stream'. Exiting)
```
Any Idea?
